### ISSUE TITLE
Create cdxj indexes for WARCs in wasCrawlDisseminationWF

### ISSUE THREAD

**Issue by lwrubel** [sul-dlss/was_robot_suite#449]
Add a cdxj-generator and cdxj-merge steps to wasCrawlDisseminationWF to create cdxj formatted indexes of incoming WARCs. These steps should go after the existing path-indexer step (so as to not delay the existing workflow steps). Those existing cdx steps will later be removed, so this should be implemented in parallel.

The [cdxj-indexer python library](https://github.com/webrecorder/cdxj-indexer) will need to be pip installed on was-robots1-* VMs as part of this work([example](https://github.com/sul-dlss/was-pywb/blob/458bbc9dd75cccafd2e702690f3db3498dbfe2b4/config/deploy.rb#L70)). Python3 is already installed via puppet. 

cdxj files will eventually be saved in `/web-archiving-stacks/data/indexes`. This is parallel to existing `/web-archiving-stacks/data/indices` directory used for openwayback's cdx files. 

Example cdxj-indexer command:
`cdxj-indexer <WARC files path> --output <path>/<filename> --sort --post-append`

wasCrawlDisseminationWF steps will need updating [on workflow-server-rails](https://github.com/sul-dlss/workflow-server-rails/blob/main/config/workflows/wasCrawlDisseminationWF.xml).

There will be subsequent tickets to scheduling the levels rolling up, and the path-indexer step for these files.

**Comment by jcoyne** [comment]
@lwrubel what should the `cdxj-generator` and `cdxj-merge` workflow steps do?

**Comment by lwrubel** [comment]
The cdx-generator should do the same steps as the current cdx-generator step. You can use that as a template. That existing step creates the CDX index files for the WARC/ARC files using the CDXGeneratorService. It generates 1 CDX file for each WARC file using jar/openwayback/bin/cdx-indexer. The generated CDX file is then copied to the cdx_working_directory as in settings. 

The new cdxj-generator step would use the cdxj-indexer script instead, still creating one index file for each WARC file. Instead of putting files in /web-archiving-stacks/data/indices/cdx_working/, settings should be added for cdxj versions of those working directories, to point to subdirectories in /web-archiving-stacks/data/indexes/

**Comment by lwrubel** [comment]
cdxj-merge is also parallel to the current cdx-sort-merge-publish step.  It merges the individual WARC cdxj files created in the previous step and creates the level0.cdx file which will live in /web-archiving-stacks/data/indexes/cdxj/

**Comment by jcoyne** [comment]
@lwrubel can you be more specific about the cdxj-merge?   Does it just need to concatenate the files into one file like this:
```
cat level[1-9].cdxj > level0.cdxj
```

**Comment by lwrubel** [comment]
The outcome should be the same as what currently exists, and you could follow the current logic or improve it if you see an opportunity to do so. The workflow step should take the single-warc cdxj files in the druid directory (created in the cdxj-generator step) and add the lines in them to a sorted level0.cdx file. The idea is that as druids go through this step, their indexed get merged into the level0.cdx. 

The logic for the rollup happens in [index.rake](https://github.com/sul-dlss/was_robot_suite/blob/main/lib/tasks/index.rake) and should still be good. The only change is that we don't want to replace the existing cdx process. We'll want to create a cdxj in the `/web-archiving-stacks/data/indexes/cdxj` directory.

**Comment by lwrubel** [comment]
BTW, the [rollup is handled in another ticket](https://github.com/sul-dlss/was_robot_suite/issues/450), not this one. That was probably confusing.

**Comment by jcoyne** [comment]
Okay, so this ticket is complete then?

**Comment by lwrubel** [comment]
We still need to fill out the [cdxj-merge](https://github.com/sul-dlss/was_robot_suite/blob/main/lib/robots/dor_repo/was_crawl_dissemination/cdxj_merge.rb) step which merges the individual WARC cdxj files into the level0.cdxj. That will unblock the next issue to do the rollup.

**Comment by jcoyne** [comment]
Oh, I didn't understand the difference between merge and roll up
